web/packages/teleterm/src/ui/AccessRequestCheckout/useAssumedRolesBar.ts

```diff
@@ -46,21 +48,18 @@ export function useAssumedRolesBar(assumedRequest: AssumedRequest) {
     getRefreshInterval(duration)
   );
 
-  const [dropRequestAttempt, dropRequest] = useAsync(() => {
-    return retryWithRelogin(
-      ctx,
-      rootClusterUri,
-      () => ctx.clustersService.dropRoles(rootClusterUri, [assumedRequest.id])
-      // TODO(gzdunek): We should refresh the resources,
-      // the same as after assuming a role in `useAssumeAccess`.
-      // Unfortunately, we can't do this because we don't have access to `ResourcesContext`.
-      // Consider moving it into `ResourcesService`.
-    ).catch(err => {
+  const [dropRequestAttempt, dropRequest] = useAsync(async () => {
+    try {
+      await retryWithRelogin(ctx, rootClusterUri, () =>
+        ctx.clustersService.dropRoles(rootClusterUri, [assumedRequest.id])
+      );
+      requestResourcesRefresh(rootClusterUri);
+    } catch (err) {
       ctx.notificationsService.notifyError({
         title: 'Could not switch back the role',
```

Comment:
i know this error text was already present but it seems so weird. I think it'd be better saying something like "Could not drop role" or "Could not switch back to default role" or something. Don't have to change here, just discussing.

Reply:
I agree, it's a little weird. I fixed it now.